src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -78,41 +95,255 @@ static DamageTable D_8099F4E8 = {
     /* UNK_DMG_0x1E   */ DMG_ENTRY(0, 0x0),
     /* Powder Keg     */ DMG_ENTRY(0, 0x0),
 };
+static ActorAnimationEntryS sAnimations[] = {
+    { &D_06004248, 1.0f, 0, -1, 0, 0 },  { &D_06004248, 1.0f, 0, -1, 0, -4 }, { &D_06003610, 1.0f, 0, -1, 0, -4 },
+    { &D_06000598, 1.0f, 0, -1, 0, -4 }, { &D_06000D48, 1.0f, 0, -1, 0, -4 }, { &D_0600219C, 1.0f, 0, -1, 0, -4 },
+    { &D_06002898, 1.0f, 0, -1, 0, -4 },
+};
+s8 D_8099F578[] = { -1, 1, 12, 13, 14, 9, 10, 11, 0, 6, 7, 8, 3, 4, 5, 2, -1, -1, -1, -1 };
+s8 D_8099F58C[] = { 0, 0, 0, 0, 3, 4, 0, 6, 7, 0, 9, 10, 0, 12, 13, 0 };
+u8 D_8099F59C[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+u16 D_8099F5AC[] = { 4000, 4, 1, 3, 6000, 4, 1, 6, 4000, 4, 1, 3, 6000, 4, 1, 6 };
+
+s32 EnZo_SetAnimation(SkelAnime* skelAnime, s16 index) {
+    s16 frameCount;
+    s32 didChange = false;
+
+    if ((index >= 0) && (index < ARRAY_COUNT(sAnimations))) {
+        didChange = true;
+        frameCount = sAnimations[index].frameCount;
+        if (frameCount < 0) {
+            frameCount = Animation_GetLastFrame(sAnimations[index].animationSeg);
+        }
+        Animation_Change(skelAnime, sAnimations[index].animationSeg, sAnimations[index].playbackSpeed,
+                         sAnimations[index].frame, frameCount, sAnimations[index].mode,
+                         sAnimations[index].transitionRate);
+    }
+    return didChange;
+}
+
+s32 EnZo_PlayWalkingSound(EnZo* this, GlobalContext* globalCtx) {
+    u8 leftWasGrounded;
+    u8 rightWasGrounded;
+    s32 waterSfxId;
+    u16 sfxId;
+    u8 isFootGrounded;
 
-#endif
+    leftWasGrounded = this->isLeftFootGrounded;
+    rightWasGrounded = this->isRightFootGrounded;
+    if (this->actor.bgCheckFlags & 0x20) {
+        if (this->actor.depthInWater < 20.0f) {
+            waterSfxId = NA_SE_PL_WALK_WATER0 - SFX_FLAG;
+        } else {
+            waterSfxId = NA_SE_PL_WALK_WATER1 - SFX_FLAG;
+        }
+        sfxId = waterSfxId + SFX_FLAG;
+    } else {
+        sfxId = SurfaceType_GetSfx(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId) + SFX_FLAG;
+    }
+    this->isLeftFootGrounded = isFootGrounded = func_8013DB90(globalCtx, &this->leftFootPos, -6.0f);
+    if ((this->isLeftFootGrounded) && (!leftWasGrounded) && (isFootGrounded)) {
+        Audio_PlayActorSound2(&this->actor, sfxId);
+    }
+    this->isRightFootGrounded = isFootGrounded = func_8013DB90(globalCtx, &this->rightFootPos, -6.0f);
+    if ((this->isRightFootGrounded) && (!rightWasGrounded) && (isFootGrounded)) {
+        Audio_PlayActorSound2(&this->actor, sfxId);
+    }
+    return 0;
```

Comment:
Again, I'd space this out a bit
```suggestion
    }

    this->isLeftFootGrounded = isFootGrounded = func_8013DB90(globalCtx, &this->leftFootPos, -6.0f);

    if ((this->isLeftFootGrounded) && (!leftWasGrounded) && (isFootGrounded)) {
        Audio_PlayActorSound2(&this->actor, sfxId);
    }

    this->isRightFootGrounded = isFootGrounded = func_8013DB90(globalCtx, &this->rightFootPos, -6.0f);
    if ((this->isRightFootGrounded) && (!rightWasGrounded) && (isFootGrounded)) {
        Audio_PlayActorSound2(&this->actor, sfxId);
    }

    return 0;
```